src/transformers/modeling_utils.py

```diff
@@ -267,7 +267,7 @@ def get_input_embeddings(self):
         else:
             raise NotImplementedError
 
-    def set_input_embeddings(self, value):
+    def set_input_embeddings(self, value: nn.Module):
```

Comment:
is this not an `nn.Layer` or even a `nn.Embedding`?

Reply:
Going through some tests, looks like it's expected to be either `nn.Embedding` or `nn.AdaptiveEmbedding`. The docstring has `nn.Module` (as "A module mapping vocabulary to hidden states"). Maybe it is so to have it be general? Not sure.